src/AzureAuth/ExceptionListToStringConverter.cs

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.Authentication.AzureAuth
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Text.Json;
+    using Microsoft.Identity.Client;
+
+    /// <summary>
+    /// The exception list to string converter.
+    /// </summary>
+    public static class ExceptionListToStringConverter
+    {
+        /// <summary>
+        /// Extracts correlation IDs from the list of exceptions.
+        /// </summary>
+        /// <param name="exceptions">List of exceptions from which correlation IDs are extracted.</param>
+        /// <returns>List of correlation IDs.</returns>
+        public static List<string> ExtractCorrelationIDsFromException(IList<Exception> exceptions)

```

Comment:
Maybe we don't need to extract these correlation IDs in exactly this way? I think we lose some information by doing so.